diff -ur '--exclude=.git' a/liveMedia/MatroskaFile.cpp b/liveMedia/MatroskaFile.cpp
--- a/liveMedia/MatroskaFile.cpp	2025-05-04 00:39:17.945145728 +0200
+++ b/liveMedia/MatroskaFile.cpp	2025-05-04 01:09:20.402439106 +0200
@@ -515,7 +515,7 @@
 }
 
 // The size of the largest key frame that we expect.  This determines our buffer sizes:
-#define MAX_KEY_FRAME_SIZE 300000
+#define MAX_KEY_FRAME_SIZE 2000000
 
 FramedSource* MatroskaFile
 ::createSourceForStreaming(FramedSource* baseSource, unsigned trackNumber,
diff -ur '--exclude=.git' a/liveMedia/RTCP.cpp b/liveMedia/RTCP.cpp
--- a/liveMedia/RTCP.cpp	2025-05-04 00:39:18.013145448 +0200
+++ b/liveMedia/RTCP.cpp	2025-05-04 01:09:20.414439076 +0200
@@ -429,7 +429,7 @@
       envir() << "RTCPInstance error: Hit limit when reading incoming packet over TCP. (fNumBytesAlreadyRead ("
 	      << fNumBytesAlreadyRead << ") >= maxRTCPPacketSize (" << maxRTCPPacketSize
 	      << ")).  The remote endpoint is using a buggy implementation of RTP/RTCP-over-TCP.  Please upgrade it!\n";
-      break;
+      exit(1);
     }
 
     unsigned numBytesRead;
